### ISSUE TITLE
MANIFEST.json and index.html

### ISSUE THREAD

**Issue by danwallach** [votingworks/arlo-e2e#7]
As part of writing everything to disk, we need to create a `MANIFEST.json` file that has the SHA256 hash of every file we write. Then, to support the command-line tools, every file we read needs to have its contents compared to the manifest, raising an error if something went wrong. The "reading" side needs to support a local filesystem as well as a remote static web host with a local cache.

While we're at it, we should write out a series of `index.html` files, such that somebody who wants to do a recursive `wget` or whatever will get everything. The top-level `index.html` should additionally pretty-print the election metadata.

**Comment by danwallach** [comment]
File naming vs. name in the manifest: the "full" file name will be platform dependent (forward vs. backward slashes), but we need a "portable" file name for the manifest.

**Comment by danwallach** [comment]
Manifest stuff seems to mostly work now. Delaying HTML for later.

**Comment by danwallach** [comment]
Need a feature for Ray distributed file writes with manifest updates. Likely API is two phase:
- Map to a tuple of file name, manifest name, and object to serialize
- Pass an array of Ray objectIDs for these tuples to a `parallel_write` method of some sort in the manifest
  - Does a `ray.remote` for the actual writing, so all the nodes will write objects out to the filesystem in parallel

**Comment by danwallach** [comment]
Distributed writing from Ray works. Still no index.html support.

**Comment by danwallach** [comment]
Closing the manifest part, making a separate issue for the web site part.
